jenkins/dockerfiles/dt-validation/Dockerfile

```diff
@@ -9,4 +9,10 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
 
 RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 500
 
-RUN pip3 install git+https://github.com/devicetree-org/dt-schema.git@master
+RUN sudo apt install flex bison
+RUN sudo apt install pkg-config python-yaml device-tree-compiler
+RUN sudo apt install libyaml-dev
```

Comment:
Please always do `apt-get update && apt-get install` in Dockerfiles, in the same `RUN` command.  This is because Docker will be caching layers based on each command, and if you have the list of packages in a separate command it won't run the "update" command which can cause some issues if the index of packages has changed.

See this example:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments

Also I think it's best to list all the packages alphabetically like in that example, with `\` to have one package per line but still only one actual 'RUN' command.  This is also what was done with the other `RUN` command on line 3.  I think however that it makes sense to keep a separate command for these packages, to clearly show they are dependencies of `dt-schema` (i.e. if these change, Docker won't have to re-install gcc-8 and vise-versa).

Reply:
+1